static/js/components/tiles/answer_table_tile.tsx

```diff
@@ -167,21 +196,55 @@ const fetchData = async (
       params: { dcids: provIdList, prop: "url" },
       paramsSerializer: stringifyFn,
     });
+    const provIdToUrl = {};
+    provIdList.forEach((provId) => {
+      const urlValues = provIdUrlResp.data[provId];
+      if (!_.isEmpty(urlValues)) {
+        provIdToUrl[provId] = new URL(urlValues[0].value).host;
+      } else {
+        provIdToUrl[provId] = "";
+      }
+    });
+
+    // Map sources to their URL
+    for (const entity in sources) {
+      for (const propertyExpr in sources[entity]) {
+        const sourceUrls = new Set<string>();
+        sources[entity][propertyExpr].forEach((provId) => {
+          sourceUrls.add(provIdToUrl[provId]);
+        });
+        sources[entity][propertyExpr] = Array.from(sourceUrls);
+      }
+    }
     return {
       entityNames,
-      sources: provIdList
-        .map((provId) => {
-          const urlValues = provIdUrlResp.data[provId];
-          if (!_.isEmpty(urlValues)) {
-            return new URL(urlValues[0].value).host;
-          } else {
-            return "";
-          }
-        })
-        .filter((url) => !!url),
+      sources,
       values,
     };
   } catch (e) {
     return null;
   }
 };
+
+/** Convert table data to a CSV for downloading */
+function generateCsv(
+  data: AnswerTableData,
+  props: AnswerTableTilePropType
+): string {
+  const headerRow = `DCID,Name,${props.columns
+    .map((column) => {
+      return `${column.header},${column.header} source`;
+    })
+    .join(",")}`;
+  const dataRows = new Array<string>();
+  props.entities.forEach((entity) => {
+    const row = [entity, data.entityNames[entity]];
+    props.columns.forEach((column) => {
+      const value = data.values[entity][column.propertyExpr].join("; ");
+      const source = data.sources[entity][column.propertyExpr].join("; ");
+      row.push(`${value},${source}`);
```

Comment:
nit: might need to check that none of the values have a comma in them & maybe enclose the value in quotes if there is a comma

Reply:
@dwnoble @chejennifer what do you think of matching the convention used in the datacommons client and encasing every string in quotes?

Reply:
That works, just make sure to escape the double quotes in that case: https://hawq.apache.org/docs/userguide/2.3.0.0-incubating/datamgmt/load/g-escaping-in-csv-formatted-files.html